src/states/joining_peer.rs

```diff
@@ -191,17 +191,20 @@ impl JoiningPeer {
 
     fn verify_signed_message(&self, msg: &SignedRoutingMessage) -> Result<(), RoutingError> {
         if let JoinType::Relocate(payload) = &self.join_type {
-            let key_info = &payload.details.content().destination_key_info;
-            msg.verify(as_iter(key_info))
-                .and_then(VerifyStatus::require_full)
-                .map_err(|error| {
-                    self.log_verify_failure(msg, &error, as_iter(key_info));
-                    error
-                })
-        } else {
-            // We don't have the root key info so we can't verify the message.
-            Ok(())
+            if let Some(details) = payload.relocate_details() {
```

Comment:
in case the payload is invalid, i.e. `relocate_details` returns `None`, this function returns with `Ok(())`, is it ok?

Reply:
Yes, essentially we have a bug if that ever return None, that mean we did not actually have a RelocateDetails in the SignedRoutingMessage, which mean we have nothing to validate trust against, and so we are like a new node here. (i.e the other JoinType).

Reply:
Yeah, maybe in that situation, we shall join as a new node? or terminate?

Reply:
I could put a log or panic here, but that bring little more, by that point we already sent the Join requests, If the message was invalid, Elder will have decided `"{} - Ignoring relocation JoinRequest from {} - invalid payload."` and not replied.

Reply:
> If the message was invalid, Elder will have decided "{} - Ignoring relocation JoinRequest from {} - invalid payload." and not replied.
Then the joining node may only get a `join request time out` error and have no idea what the problem is, and keeps re-join using the same info ? hmm...   I think a bit more detailed failure reason shall be given?

Reply:
But then, the bug is worth than that, it means a group of valid Elders sent an invalid message.
Essentially the problem is the type SignedRoutingMessage does not embed all the type information. (i.e that as an Adult we would never have reached the Relocate state if the message was not a Relocate message. One alternative is to store both RelocateDetails and SignedRoutingMessage, thus keeping type information but now getting duplicate info. Another alternative is to have a SignedRelocateMessage that contiain a SignedRoutingMessage that had to contain RelocateDetails, I can see that as a reasonable approach, I'll look at adding a new commit that would remove that confusion.